samza-sql-api/src/main/java/org/apache/samza/sql/interfaces/udfs/ScalarUdfContext.java

```diff
@@ -0,0 +1,56 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.interfaces.udfs;
+
+import java.util.Properties;
+
+
+/**
+ * The context that is accessible to UDFs, Any configs that the UDFs require to execute
+ * will be accessible through this UDFContext.
+ */
+public class ScalarUdfContext {
+
+  /**
+   * Since we don't control the instantiation of the UDF classes as well as
+   * invocation of the execute methods on them. We use singleton pattern here to pass the context to the UDF.
+   */
+  private static ScalarUdfContext instance;
+
+  /**
+   * This contains all the samza configs including the samza system configs.
+   */
+  private static Properties config;
+
+  private ScalarUdfContext() {
+  }
+
+  private static void createInstance(Properties properties) {
+    config = properties;
+  }
+
+  public static ScalarUdfContext getInstance() {
```

Comment:
can you remove the "static" declaration here. Seems that is better to use getInstance().getConfig()